content/docs/querying/operators.md

```diff
@@ -50,31 +58,101 @@ either `0` (`false`) or `1` (`true`), depending on the comparison result.
 value of every data sample in the vector, and vector elements between which the
 comparison result is `false` get dropped from the result vector.
 
-**Between two instant vectors**, these operators behave as a filter: They apply
-to vector elements that have identical sets of labels between the two vectors.
-Vector elements for which the expression evaluates to `false` or which don't
-find an exact label match on the other side of the expression get dropped from
-the result, while the others get carried over into a result vector with their
-original (left-hand-side) metric names and data values.
+**Between two instant vectors**, these operators behave as a filter, applied to
+matching entries. Vector elements for which the expression is not true or
+which do not find a match on the other side of the expression get dropped from the
+result, while the others are propagated into a result vector with their original
+(left-hand-side) metric names and label values.
 
-# Logical/set binary operators
+### Logical/set binary operators
 
 These logical/set binary operators are only defined between instant vectors:
 
 * `and` (intersection)
-* `or` (union) (**not yet implemented!**)
+* `or` (union)
 
 `vector1 and vector2` results in a vector consisting of the elements of
 `vector1` for which there are elements in `vector2` with exactly matching
-labelsets. Other elements are dropped. The metric name and values are carried
+label sets. Other elements are dropped. The metric name and values are carried
 over from the left-hand-side vector.
 
 `vector1 or vector2` results in a vector that contains all original elements
-(labelsets + values) of `vector1` and additionally all elements of `vector2`,
-which don't have matching labelsets in `vector1`. The metric name is carried
-over from the left-hand-side vector in both cases.
+(label sets + values) of `vector1` and additionally all elements of `vector2`,
+which do not have matching label sets in `vector1`.
+
+## Vector matching
+
+Operations between vectors attempt to find a matching element in the right-hand-side
+vector for each entry in the left-hand-side. There are two basic types of
```

Comment:
s/left-hand-side/left-hand side/ in this case because "side" is not part of the compound adjective anymore here, but the described noun itself (http://www.gingersoftware.com/content/grammar-rules/adjectives/compound-adjectives/).